### PR TITLE
Fix bibtex and warnings

### PR DESCRIPTION
**Pull request by Akshobhya1234**
Include all information that is available for a reference.  This PR removes all warning messages.

### CODE REVIEW

File: references.bib [hunk 1]
```diff
@@ -207,11 +213,13 @@ @techreport{kang1990feature
     year        = {1990},
     institution = {Carnegie-Mellon Univ Pittsburgh Pa Software Engineering Inst}
   }
-  @inproceedings{hu2021formalizing,
+  @misc{hu2021formalizing,
```
Comment (JacquesCarette) [inline references.bib]: You should not be using 'misc' to much - this is a hacky way to fix things. This is a proceedings paper, you should leave it like that.

File: references.bib [hunk 2]
```diff
@@ -51,22 +51,20 @@ @inbook{Paulin-Mohring2012
...
+    year          = {2019},
+    month         = {December},
     archiveprefix = {arXiv},
     eprint        = {1912.03028},
     primaryclass  = {cs.SE},
     adsurl        = {https://ui.adsabs.harvard.edu/abs/2019arXiv191203028S},
     adsnote       = {Provided by the SAO/NASA Astrophysics Data System}
   }
-  @book{assia_mahboubi_2021_4457887,
+  @misc{assia_mahboubi_2021_4457887,
```
Comment (JacquesCarette) [inline references.bib]: Others use
```
@article{mahboubi2021mathematical,
  title={Mathematical components},
  author={Mahboubi, Assia and Tassi, Enrico},
  journal={Online book},
  year={2021}
}
```
which is better

File: references.bib [hunk 3]
```diff
@@ -165,18 +167,22 @@ @article{kunen1996moufang
...
+    author = {Deng, Fang'an and Chen, Lu and Tuo, Shouheng and Ren, ShengZhang},
+    year = {2016},
+    month = {01},
+    pages = {1-7},
+    title = {Characterizations of N(2,2,0) Algebras},
+    volume = {2016},
+    journal = {Algebra},
+    doi = {10.1155/2016/2752681}
+  }
+  @misc{didurik2018some,
```
Comment (JacquesCarette) [inline references.bib]: Surely you can find something actually published instead of a preprint for this material?

File: references.bib [hunk 4]
```diff
@@ -223,12 +231,18 @@ @article{sankappanavar1981course
     year      = {1981},
     publisher = {Citeseer}
   }
-  @book{russell2020principles,
-    title     = {The Principles Of Mathematics},
-    author    = {Russell, Bertrand},
-    year      = {2020},
-    publisher = {Routledge}
+
+  @article{russell2020principles,
```
Comment (JacquesCarette) [inline references.bib]: This citation is definitely not right - the author needs to be Bertrand Russell! What was wrong with the previous one, which seemed much better.

File: references.bib [hunk 5]
```diff
@@ -366,11 +384,11 @@ @misc{barnett2017roots
     url   = {https://digitalcommons.ursinus.edu/cgi/viewcontent.cgi?article=1002&context=triumphs_abstract}
   }
 
-  @Book{plfa22.08,
+  @misc{plfa22.08,
```
Comment (JacquesCarette) [inline references.bib]: No, please use @book.  See https://www.bibtex.com/e/book-entry/

File: references.bib [hunk 6]
```diff
@@ -398,7 +416,7 @@ @misc{sannella2012foundations
     publisher={Springer Science \& Business Media}
   }
 
-  @book{gries2013logical,
+  @misc{gries2013logical,
```
Comment (JacquesCarette) [inline references.bib]: No! This is a book, don't hack like this.

File: references.bib [hunk 7]
```diff
@@ -411,25 +429,32 @@ @misc{musa
     year   = {2021},
     url    = {http://hdl.handle.net/11375/26373},
   }
-  @book{wechler2012universal,
-    title={Universal Algebra For Computer Scientists},
-    author={Wechler, Wolfgang},
-    volume={25},
+
+  @misc{wechler2012universal,
```
Comment (JacquesCarette) [inline references.bib]: same here, don't hack.

File: references.bib [hunk 8]
```diff
@@ -411,25 +429,32 @@ @misc{musa
...
     author = "{Agda Development Team}",
     url   = {https://agda.readthedocs.io/en/v2.6.3/getting-started/what-is-agda.html},
     year  = {2023}
   } 
-  @book{AgdaTB,
-    title={Verified Functional Programming In Agda},
-    author={Stump, Aaron},
-    year={2016},
-    publisher={Morgan \& Claypool}
+  @misc{AgdaTB,
```
Comment (JacquesCarette) [inline references.bib]: No! It is a published book.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
I stopped reviewing after a while -- you just hacked things up. Please fix this properly.

**Comment by Akshobhya1234** [comment]
> I stopped reviewing after a while -- you just hacked things up. Please fix this properly.

I have fixed them. Now `@misc` is used for arXiv and online documentation.

File: references.bib [hunk 9]
```diff
@@ -519,15 +544,15 @@ @misc{abel2012agda
   @misc{danielsson2011agda,
```
Comment (JacquesCarette) [inline references.bib:543]: That is not the correct bib entry for the standard library!

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Just one fix to go.

File: references.bib (hunk 9 above)
Comment (Akshobhya1234) [reply references.bib:543]: I had another reference for the standard library. I fixed them now and kept the version as v1.7.

**Review by JacquesCarette — APPROVED** [review]

**Comment by JacquesCarette** [comment]
You're going to have to do a merge by hand because there are conflicts now.

**Pull request merged**